### ISSUE TITLE
[Bug][plugins/dora] `incident_deploy_connector` is not matching incident with deployments properly

### ISSUE THREAD

**Issue by Adibov** [apache/devlake#8150]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar issues.

### What happened

DORA has introduced a patch to the definition of CFR in its 2023 report:
> For the primary application or
service you work on, how long does it generally take
to restore service **after a change to production or
release to users results in degraded service (for
example, lead to service impairment or service
outage) and subsequently require remediation
(for example, require a hotfix, rollback, fix
forward, or patch)**

The definition has changed to only consider those downtimes as incidents, caused by a deployment. That is for example, if we have two consecutive deployments with 12 hours between them, and some incidents happened after 6 hours after the first deployment, we shouldn't count that incident as a change that led to a failure.

However, in the DORA plugin in Devlake, we match any incident with the latest deployment before it, ignoring the time length between them. This leads to imprecise metrics.

### What do you expect to happen

I think a preferred solution is to configure the time length between a deployment and an incident, which we consider as a software fault. That is, having some variable in code that is configurable and user can configure it based on their needs.

### How to reproduce

Create one deployment and an incident far after it, that incident will match the deployment.

### Version

v1.0.1@e061ef2

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by Shaktty** [comment]
I don't agree here. An incident can take time to appear or manifest itslef, but still be due to a deployment of a buggy artifact with defect leakeage. Also, not only full downtime should be identified as a faulty deployment, but also any incident that states a degraded user experience.

**Comment by Adibov** [comment]
@Shaktty 
I'm on the same page with you, software defects must also include defect leakage and not only full downtimes. That is why I proposed a configurable window to only consider those incidents lying in that window. For example, suppose we deploy a manifest at 6 PM and its availability drops to 90% at 6 AM the next morning. In that case, we shouldn't consider it an incident since it's probably because of an infrastructure defect and is not software-related.

Another possible example is that when the whole Kubernetes cluster goes down, all availabilities drop, and if we submit an incident via webhook, it will be considered a software incident but it's not.
